pgjdbc/src/test/java/org/postgresql/test/jdbc2/BinaryTest.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.jdbc2;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.postgresql.core.Oid;
+import org.postgresql.core.QueryExecutor;
+import org.postgresql.jdbc.PgConnection;
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PGBinaryObject;
+import org.postgresql.util.PGobject;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * TestCase to test handling of binary types.
+ */
+public class BinaryTest {
+  private Connection con;
+
+  /**
+   * Set up the fixture for this testcase: the tables for this test.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @Before
+  public void setUp() throws SQLException {
+    con = TestUtil.openDB();
+
+    TestUtil.createTable(con, "test_binary", "id integer,name text,geom point");
+
+    TestUtil.closeDB(con);
+  }
+
+  /**
+   * Tear down the fixture for this test case.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @After
+  public void tearDown() throws SQLException {
+    TestUtil.closeDB(con);
+
+    con = TestUtil.openDB();
+
+    TestUtil.dropTable(con, "test_binary");
+
+    TestUtil.closeDB(con);
+  }
+
+  /**
+   * Make sure the functions for adding binary transfer OIDs for custom types are correct.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @Test
+  public void testBinaryTransferOids() throws SQLException {
+    con = TestUtil.openDB();
+    QueryExecutor queryExecutor = ((PgConnection) con).getQueryExecutor();
```

Comment:
`PgConnection` is driver-internal API. Client-facing APIs are `java.sql.Connection` and `org.postgresql.PGConnection`.

Reply:
This is also a driver-internal test. The APIs for checking if a type is sent or received as binary are only available on `QueryExecutor`. So they can only be tested by either really sending and receiving them (what `testCustomBinaryTypes`) does or using the `QueryExecutor` API.
I added `testBinaryTransferOids` by intention to make sure the internal API works before a database is used in the other test. So it is possible to first test the API works and then to test if the database access works.